pkg/expansion/graph.go

```diff
@@ -0,0 +1,116 @@
+package expansion
+
+import (
+	"fmt"
+
+	expansionunversioned "github.com/open-policy-agent/gatekeeper/apis/expansion/unversioned"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+type cycleDetector interface {
+	addTemplate(*expansionunversioned.ExpansionTemplate) error
+	removeTemplate(*expansionunversioned.ExpansionTemplate)
+}
+
+// gvkGraph represents a directed acyclic graph of GVKs. The terminal int value
+// represents how many edges exists between two GVKs.
```

Comment:
let's also define the edge between the nodes, ie what it means for two GVKs to have an edge.